gammapy/estimators/utils.py

```diff
@@ -770,83 +773,350 @@ def combine_flux_maps(maps, method="gaussian_errors", reference_model=None):
+        Default is None and is will use the reference_model of the first FluxMaps in the list.
+
+    dnde_scan_axis : `~gammapy.maps.MapAxis`
+        dnde axis used to compute the likelihood profile.
+        Used only if `method` is distrib or profile.
+        Default is None and it will be derived from the first FluxMaps in the list.
 
     Returns
     -------
     flux_maps : `~gammapy.estimators.FluxMaps`
         Joint flux map.
     """
-    if method != "gaussian_errors":
-        raise ValueError(
-            f'Invalid method : {method}, available methods are : "gaussian_errors"'
-        )
 
-    if isinstance(maps, FluxMaps):
-        geom = maps.dnde.sum_over_axes().geom
-        means = list(maps.dnde.copy().iter_by_image(keepdims=True))
-        sigmas = list(maps.dnde_err.copy().iter_by_image(keepdims=True))
-        mean = Map.from_geom(geom, data=means[0].data, unit=means[0].unit)
-        sigma = Map.from_geom(geom, data=sigmas[0].data, unit=sigmas[0].unit)
-        gti = maps.gti
-        meta = maps.meta
-        if reference_model is None:
-            reference_model = maps.reference_model
+    gtis = [map_.gti for map_ in maps if map_.gti is not None]
+    if np.any(gtis):
+        gti = gtis[0].copy()
+        for k in range(1, len(gtis)):
+            gti.stack(gtis[k])
     else:
+        gti = None
+    # TODO : change this once we have stackable metadata objets
+    metas = [map_.meta for map_ in maps if map_.meta is not None]
+    meta = {}
+    if np.any(metas):
+        for data in metas:
+            meta.update(data)
+    if reference_model is None:
+        reference_model = maps[0].reference_model
+
+    if method == "gaussian_errors":
+
         means = [map_.dnde.copy() for map_ in maps]
         sigmas = [map_.dnde_err.copy() for map_ in maps]
+        # compensate for the ts deviation from gaussian approximation expectation in each map
+        ts_diff = np.nansum(
+            [
+                map_.ts.data - (map_.dnde.data / map_.dnde_err.data) ** 2
+                for map_ in maps
+            ],
+            axis=0,
+        )
+
         mean = means[0]
         sigma = sigmas[0]
 
-        gtis = [map_.gti for map_ in maps if map_.gti is not None]
-        if np.any(gtis):
-            gti = gtis[0].copy()
-            for k in range(1, len(gtis)):
-                gti.stack(gtis[k])
-        else:
-            gti = None
+        for k in range(1, len(means)):
+
+            mean_k = means[k].quantity.to_value(mean.unit)
+            sigma_k = sigmas[k].quantity.to_value(sigma.unit)
+
+            mask_valid = np.isfinite(mean) & np.isfinite(sigma) & (sigma.data != 0)
+            mask_valid_k = np.isfinite(mean_k) & np.isfinite(sigma_k) & (sigma_k != 0)
+            mask = mask_valid & mask_valid_k
+            mask_k = ~mask_valid & mask_valid_k
+
+            mean.data[mask] = (
+                (mean.data * sigma_k**2 + mean_k * sigma.data**2)
+                / (sigma.data**2 + sigma_k**2)
```

Comment:
A weighted sum of the means is realized here. The used weights are w_i = sigma_i**2, ie larger is the sigma more weight is given to the summed mean. Is it intentional? Spontaneously, I would say that the weights should be w_i = 1/sigma_i**2...

Reply:
I used these relations : https://ccrma.stanford.edu/~jos/sasp/Product_Two_Gaussian_PDFs.html

Reply:
Ah yeah, I missed that. Thanks.